src/Visitor/ReflectionVisitor.php

```diff
@@ -184,4 +182,26 @@ private function isFunctionLikeNode(Node $node): bool
 
         return false;
     }
+
+    private function getClassReflectionForNode(Node\Stmt\ClassLike $node): ClassReflection
+    {
+        /** @var Node\Name|null $fqn */
+        $fqn = $node->getAttribute(FullyQualifiedClassNameVisitor::FQN_KEY);
+
+        if ($fqn instanceof Node\Name) {
+            return CoreClassReflection::fromClassName($fqn->toString());
+        }
+
+        assert($node instanceof Node\Stmt\Class_);
```

Comment:
I'm not sure to understand this assert neither how we get from `ClassLike` to `Class_` here... Is it based on the existence of the name?

Reply:
If the classlike has no name then it is an anonymous class. This is the best way i could think of to make that clear to our tooling etc.